crypto/store/loader_file.c

```diff
@@ -764,20 +764,22 @@ static OSSL_STORE_LOADER_CTX *file_open(const OSSL_STORE_LOADER *loader,
         }
     } else {
         BIO *buff = NULL;
-        char peekbuf[4096];
+        char *peekbuf = OPENSSL_malloc(4096);
 
-        if ((buff = BIO_new(BIO_f_buffer())) == NULL
+        if (peekbuf == NULL || (buff = BIO_new(BIO_f_buffer())) == NULL
```

Comment:
The check of peekbuf should be separate and generate an error, like this:

``` C
        if (peekbuf == NULL) {
            OSSL_STOREerr(OSSL_STORE_F_FILE_OPEN, ERR_R_MALLOC_FAILURE);
            goto err;
        }
```

The check of `buff` doesn't have the same requirements, since `BIO_new` will generate an error of its own when it fails.